### PR TITLE
Modern gradle

### PR DESCRIPTION
**Pull request by lesserwhirls**
This PR builds supersedes SpectraLogic/ds3_java_sdk#595. Starting with the work @scribe did, the new default behavior of jackson xml handling for empty xml elements was reverted to the previous default behavior and the tests passed. From there, more dependencies were updated, and the gradle build was moved from the groovy DSL to the kotlin DSL. Finally, gradle was upgraded from 7.5 to 7.5.1.

This work should result in the same build outputs and behavior as the previous build, with the exclusion of the configuration for artifact publishing, which was still configured to use bintray (will add publishing back in another PR to use GitHub packages).

### CODE REVIEW

**Review by scribe — APPROVED** [review]
Looks good! I'll close my PR so there's no confusion.

File: ds3-sdk/src/test/java/com/spectralogic/ds3client/helpers/RequestMatchers.java [hunk 1]
```diff
@@ -23,16 +23,15 @@
 import com.spectralogic.ds3client.commands.spectrads3.GetJobChunksReadyForClientProcessingSpectraS3Request;
 import com.spectralogic.ds3client.models.JobChunkClientProcessingOrderGuarantee;
 import org.apache.commons.io.IOUtils;
-import org.hamcrest.Description;
-import org.hamcrest.TypeSafeMatcher;
-import org.mockito.ArgumentMatcher;
+import org.hamcrest.*;
```
Comment (david-limbach) [inline ds3-sdk/src/test/java/com/spectralogic/ds3client/helpers/RequestMatchers.java:26]: Can we avoid the * import?

**Review by david-limbach — APPROVED** [review]

File: ds3-sdk/src/test/java/com/spectralogic/ds3client/helpers/RequestMatchers.java (hunk 1 above)
Comment (lesserwhirls) [reply ds3-sdk/src/test/java/com/spectralogic/ds3client/helpers/RequestMatchers.java]: Yep - I caught a few more of them that were added.

**Review by scribe — APPROVED** [review]
👍

**Comment by lesserwhirls** [comment]
Travis keeps getting hung up running the test:

~~~
>80% EXECUTING [47s]> :ds3-sdk:test
>Executing test com.spectralogic.ds3client.helpers.FileObjectG
> :ds3-sdk:test
> 155 tests completed, 2 skipped<==========---
> 80% EXECUTING [48s]<==========---
~~~

I was able to run locally no problem. Travis is running a release of Oracle JDK from 2017 (1.8.0_151), and that might not be playing nice with the latest version of gradle (I'm running Temurin build 1.8.0_332-b09 locally).

What I could do is make a new PR against current master to move to github actions for testing, and if that is smooth, then update this PR to use that. That will at least get us a more recent version of JDK 8 to work with.

**Review by david-limbach — APPROVED** [review]

**Review by david-limbach — APPROVED** [review]

**Review by scribe — APPROVED** [review]

**Pull request merged**